tests/sgx_zerobase/host/host.cpp

```diff
@@ -44,48 +58,142 @@ int main(int argc, const char* argv[])
 
     result = oe_create_sgx_zerobase_enclave(
         argv[1], OE_ENCLAVE_TYPE_SGX, flags, NULL, 0, &enclave);
     if (result != OE_OK)
     {
-        fprintf(stderr, "Could not create enclave, result=%d\n", result);
+        fprintf(stderr, "Could not create 0-base enclave, result=%d\n", result);
         return 1;
     }
-    char output[1024];
-    const char* source = "My First App\n";
+
+    fprintf(
+        stdout,
+        "\nTest 3: Test ecall, ocall on 0-base enclave\n"
+        "Expected result : OE_OK\n");
+
+    const char* input = "testing ecall\0";
     int res = -1;
-    OE_TEST(
-        secure_string_patching(
-            enclave, &res, source, output, OE_COUNTOF(output)) == OE_OK);
+
+    OE_TEST(test_ecall(enclave, &res, input) == OE_OK);
 
     if (res != 0)
     {
-        fprintf(stderr, "%s: enclave called failed\n", argv[0]);
-        exit(1);
+        fprintf(stderr, "[host]: ecall/ocall failed %d\n", res);
+        return 1;
+    }
+
+    fprintf(
+        stdout,
+        "\nTest 4: Check host access to [/proc/sys/vm/mmap_min_addr, "
+        "elrange->image_start_address]\n"
+        "Expected result : success\n");
+
+    uint64_t address = 0x15000;
+
+    if (test_host_mmap(address) != 0)
+    {
+        fprintf(stderr, "host mmap at address %lx failed\n", address);
+        return 1;
     }
 
-    const char expect[] = "My First App\n"
-                          "Hello world from Enclave\n"
-                          "My First App\n"
-                          "Hello world from Host\n";
+    fprintf(
+        stdout,
+        "\nTest 5: Check host access at address 0x0\n"
+        "Expected result : fail\n");
 
-    if (strcmp(output, expect) != 0)
+    address = 0x0;
+
+    if (test_host_mmap(address) == 0)
     {
-        fprintf(stderr, "%s: returned string don't match\n", argv[0]);
+        fprintf(stderr, "host mmap at address %lx succeeded\n", address);
         return 1;
     }
 
+    /* Enclave memory access tests */
+    if (flags & OE_ENCLAVE_FLAG_SIMULATE)
+        fprintf(
+            stdout,
+            "Simulation mode does not support exceptions. Skip the test "
+            "ECALL.\n");
+    else if (_is_misc_region_supported())
+    {
+        struct exceptions exception;
+
+        fprintf(
+            stdout,
+            "\nTest 6: Enclave memory access at 0x0\n"
+            "Expected result: page fault\n");
+
+        address = 0x0;
+
+        OE_TEST(test_enclave_memory_access(enclave, &res, address, &exception));
+        if (res != 0)
+        {
+            fprintf(stderr, "test_enclave_memory_access failed %d\n", res);
+            return 1;
+        }
+
+        fprintf(stdout, "address %lx, exception %d\n", address, exception.code);
+        if (!exception.exception)
+            return 1;
+
+        fprintf(
+            stdout,
+            "\nTest 7: memory access at 0x22000 in a 0-base enclave\n"
+            "Expected result: no exceptions\n");
+
+        address = 0x30000;
```

Comment:
Add comment or print the 0-base settings and explain why access to this address does not cause fail